src/main/java/io/strimzi/kproxy/KafkaProxy.java

```diff
@@ -128,10 +132,24 @@ public KafkaProxy startup() throws InterruptedException {
         KafkaProxyInitializer initializer = new KafkaProxyInitializer(brokerHost, brokerPort, interceptorProviderFactory, logNetwork, logFrames);
 
         // Configure the bootstrap.
-        bossGroup = new NioEventLoopGroup(1);
-        workerGroup = new NioEventLoopGroup();
+        final Class<? extends ServerChannel> channelClass;
+        if (Epoll.isAvailable()) {
+            bossGroup = new EpollEventLoopGroup(1);
+            workerGroup = new EpollEventLoopGroup();
+            channelClass = EpollServerSocketChannel.class;
+        }
+        else if (KQueue.isAvailable()) {
+            bossGroup = new KQueueEventLoopGroup(1);
+            workerGroup = new KQueueEventLoopGroup();
+            channelClass = KQueueServerSocketChannel.class;
+        }
+        else {
+            bossGroup = new NioEventLoopGroup(1);
+            workerGroup = new NioEventLoopGroup();
+            channelClass = NioServerSocketChannel.class;
+        }
         ServerBootstrap serverBootstrap = new ServerBootstrap().group(bossGroup, workerGroup)
-                .channel(NioServerSocketChannel.class)
+                .channel(channelClass)
```

Comment:
@franz1981 so IIUC, this will mean we _have_ to expose Netty's `ByteBuf` to our handlers, right? Rather than NIO `ByteBuffer`, which so far I had been hoping to do, so as to avoid exposing Netty to handlers at all.